test/foundry/src/concrete/receipt/Receipt.t.sol

```diff
@@ -0,0 +1,253 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.25;
+
+import {Test, Vm} from "forge-std/Test.sol";
+import "forge-std/console.sol";
+import {Receipt} from "../../../../../contracts/concrete/receipt/Receipt.sol";
+import {IReceiptOwnerV1} from "../../../../../contracts/interface/IReceiptOwnerV1.sol";
+import {TestReceipt} from "../../../../../contracts/test/TestReceipt.sol";
+import {TestReceiptOwner} from "../../../../../contracts/test/TestReceiptOwner.sol";
+
+contract ReceiptTest is Test {
+    event ReceiptInformation(address sender, uint256 id, bytes information);
+
+    function generateNonEmptyBytes(uint256 maxLength, uint256 seed) internal pure returns (bytes memory) {
```

Comment:
maybe replace with fuzzing bytes and truncating if needed with https://github.com/rainlanguage/rain.solmem/blob/main/src/lib/LibBytes.sol#L16